[Az.DevCenter] Update control plane and data plane APIs to v2023-10-01-preview

am-lim opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

Update control plane and data plane APIs to 2023-10-01-preview versions. 
Design reviews:
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1384
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1389

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

NoriZC commented on src/DevCenter/DevCenter.AutoRest/custom/New-AzDevCenterAdminPool.ps1:
```diff
@@ -111,60 +111,66 @@ function New-AzDevCenterAdminPool {
     
-        [Parameter(ParameterSetName = 'CreateExpanded', Mandatory)]
-        [Parameter(ParameterSetName = 'CreateViaIdentityExpanded', Mandatory)]
+        [Parameter(Mandatory)]
```
By expanding the parametersets where the Location must be Mandatory, there will be breaking changes. Do you know the context?

NoriZC commented:
Hi @am-lim, as we renamed the parameters, there are realated breaking changes detected in our CI pipelines. Please add alias for the parameters as mentioned in the details. Here is an example:
https://github.com/Azure/azure-powershell/blob/58ef510d16e82f4e31bd98067e234b5656f206fc/src/MySql/MySql.Autorest/README.md?plain=1#L206-L211

am-lim commented:
> Hi @am-lim, as we renamed the parameters, there are realated breaking changes detected in our CI pipelines. Please add alias for the parameters as mentioned in the details. Here is an example:
> 
> https://github.com/Azure/azure-powershell/blob/58ef510d16e82f4e31bd98067e234b5656f206fc/src/MySql/MySql.Autorest/README.md?plain=1#L206-L211

Hi @NoriZC I updated the read me but still seeing the error, this is because "DevCenter" is a parameter that we add as a customization and is not part of the generated cmdlets. How do we fix this?

NoriZC commented:
> > Hi @am-lim, as we renamed the parameters, there are realated breaking changes detected in our CI pipelines. Please add alias for the parameters as mentioned in the details. Here is an example:
> > https://github.com/Azure/azure-powershell/blob/58ef510d16e82f4e31bd98067e234b5656f206fc/src/MySql/MySql.Autorest/README.md?plain=1#L206-L211
> 
> Hi @NoriZC I updated the read me but still seeing the error, this is because "DevCenter" is a parameter that we add as a customization and is not part of the generated cmdlets. How do we fix this?

Sorry I didn't notice that. Please directly update the custom file.

am-lim commented:
> > > Hi @am-lim, as we renamed the parameters, there are realated breaking changes detected in our CI pipelines. Please add alias for the parameters as mentioned in the details. Here is an example:
> > > https://github.com/Azure/azure-powershell/blob/58ef510d16e82f4e31bd98067e234b5656f206fc/src/MySql/MySql.Autorest/README.md?plain=1#L206-L211
> > 
> > 
> > Hi @NoriZC I updated the read me but still seeing the error, this is because "DevCenter" is a parameter that we add as a customization and is not part of the generated cmdlets. How do we fix this?
> 
> Sorry I didn't notice that. Please directly update the custom file.

Hi @NoriZC why is the current build failing? Tried debugging, but not sure. I fixed the instances of DevCenter to DevCenterName

NoriZC commented:
Hi @am-lim I have resolved the errors introduced by the renaming, and the build succeeded. Please work upon this PR to resolve the issues left.

am-lim commented:
> Hi @am-lim I have resolved the errors introduced by the renaming, and the build succeeded. Please work upon this PR to resolve the issues left.

Thank you @NoriZC. I want to remove two parameters from "Update-AzDevCenterAdminPool", this is a breaking change. How do we allow this breaking change?

NoriZC commented:
Hi @am-lim For GAed modules, we don't allow breaking changes unless in major releases. The major release will occur once half year, next one will be May 2024. What you can do in this sprint is to announce the breaking change. You can follow [this link](https://github.com/Azure/azure-powershell/blob/9e28914b54b6793548e31407c6f61d474110e776/documentation/development-docs/breakingchange-for-autogen-module.md#case-2--breaking-change-for-a-cmdlet) to add a warning message annoucing this issue.

wyunchi-ms commented:
/azp run

am-lim commented:
> Hi @am-lim For GAed modules, we don't allow breaking changes unless in major releases. The major release will occur once half year, next one will be May 2024. What you can do in this sprint is to announce the breaking change. You can follow [this link](https://github.com/Azure/azure-powershell/blob/9e28914b54b6793548e31407c6f61d474110e776/documentation/development-docs/breakingchange-for-autogen-module.md#case-2--breaking-change-for-a-cmdlet) to add a warning message annoucing this issue.

Thanks @NoriZC ! Will update this in another version. The PR is now ready for review again.
